src/engine/Minus.cpp

```diff
@@ -145,29 +151,31 @@ IdTable Minus::computeMinus(
     keepEntry.at(ql::ranges::distance(joinColumnsLeft.begin(), leftIt)) = false;
   };
 
+  auto handleCompatibleRow = [&markForRemoval](const auto& leftIt,
+                                               const auto& rightIt) {
+    const auto& leftRow = *leftIt;
+    const auto& rightRow = *rightIt;
+    bool onlyMatchesBecauseOfUndef = ql::ranges::all_of(
+        ::ranges::views::zip(leftRow, rightRow), [](const auto& tuple) {
+          const auto& [leftId, rightId] = tuple;
+          return leftId.isUndefined() || rightId.isUndefined();
+        });
+    if (!onlyMatchesBecauseOfUndef) {
+      markForRemoval(leftIt);
+    }
+  };
+
   std::visit(
-      [this, &joinColumnsLeft, &joinColumnsRight, &markForRemoval](
+      [this, &joinColumnsLeft, &joinColumnsRight,
+       handleCompatibleRow = std::move(handleCompatibleRow)](
           auto findUndefLeft, auto findUndefRight) {
         [[maybe_unused]] auto outOfOrder = ad_utility::zipperJoinWithUndef(
```

Comment:
You can remove the `[[maybe_unused]]` and `CORRECTNESS_CHECK` that it is in fact not out of order:)

Reply:
As I explained, they can be out of order, but it doesn't matter since inserting into `std::vector<bool>` which is pre-filled implicitly sorts the values anyway.